bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BalRuntime.java

```diff
@@ -327,36 +341,104 @@ public void registerStopHandler(BFunctionPointer<?, ?> stopHandler) {
     }
 
     private void invokeConfigInit() {
-        String configClassName = getConfigClassName(this.module);
-        Class<?> configClazz;
-        try {
-            configClazz = Class.forName(configClassName);
-        } catch (Throwable e) {
-            throw ErrorCreator.createError(StringUtils.fromString("failed to load configuration class :" +
-                    configClassName));
-        }
+        Class<?> configClass = loadClass(CONFIGURATION_CLASS_NAME);
         ConfigDetails configDetails = LaunchUtils.getConfigurationDetails();
         String funcName = Utils.encodeFunctionIdentifier("$configureInit");
         try {
-            final Method method = configClazz.getDeclaredMethod(funcName, String[].class, Path[].class, String.class);
+            final Method method = configClass.getDeclaredMethod(funcName, String[].class, Path[].class, String.class);
             method.invoke(null, new String[]{}, configDetails.paths, configDetails.configContent);
         } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
             throw ErrorCreator.createError(StringUtils.fromString("configurable initialization failed due to " +
                     RuntimeUtils.formatErrorMessage(e)), e);
         }
     }
 
-    private static String getConfigClassName(Module module) {
-        String configClassName = CONFIGURATION_CLASS_NAME;
+    private void invokeModuleStop() {
+        Class<?> initClass = loadClass(MODULE_INIT_CLASS_NAME);
+        String funcName = Utils.encodeFunctionIdentifier("$moduleStop");
+        try {
+            final Method method = initClass.getDeclaredMethod(funcName, RuntimeRegistry.class);
+            method.invoke(null, scheduler.getRuntimeRegistry());
+
+        } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
+            throw ErrorCreator.createError(StringUtils.fromString("calling module stop failed due to " +
```

Comment:
```suggestion
            throw ErrorCreator.createError(StringUtils.fromString("failed to stop module due to " +
```
Not sure if this is the intention of the message but this looks cleaner.

Reply:
Fixed